crowbar_framework/app/models/api/node.rb

```diff
@@ -125,6 +125,32 @@ def post_upgrade
       end
     end
 
+    # Reboot the node and wait until it comes back online
+    def reboot_and_wait
+      ssh_status = @node.ssh_cmd("/sbin/reboot")
+      if ssh_status[0] != 200
+        save_error_state("Failed to reboot the machine. Could not ssh.")
+        return false
+      end
+
+      begin
+        Timeout.timeout(300) do
+          loop do
+            ssh_status = @node.ssh_cmd("")
+            break if ssh_status[0] == 200
```

Comment:
I think you have to first wait for the `reboot` to kill `sshd`, and only then wait for `sshd` to come back post-reboot.  Otherwise I can imagine this passing straight through even before the node has rebooted.